### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by krishna-ku**

### CODE REVIEW

File: src/main/java/com/restaurant/controller/RatingController.java [hunk 1]
```diff
@@ -33,10 +33,10 @@ public class RatingController {
 	 * @param RatingDto
 	 * @return RatingDto {@link com.restaurant.dto.RatingDto}
 	 */
-	@PostMapping("/{id}")
-	public ResponseEntity<RatingDto> createRating(@Valid @RequestBody RatingDto ratingDto, @PathVariable long id) {
+	@PostMapping("/{orderId}/{userId}")
```
Comment (crack-byte) [inline src/main/java/com/restaurant/controller/RatingController.java]: The path should be meaningful. 
Example:- 
/{orderId}/user/{userId}

File: src/main/java/com/restaurant/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -33,13 +33,13 @@ public UserDto createUser(UserDto userDto) {
 
 		User user1 = userRepo.findByEmail(userDto.getEmail());// validate by pattern matcher
 
-		if (user1 == null) {
+		if (user1 == null && Keywords.EMAIL_REGEX.matcher(userDto.getEmail()).matches()) {
```
Comment (crack-byte) [inline src/main/java/com/restaurant/service/impl/UserServiceImpl.java]: check regex match before finding by email in the database.

File: src/main/java/com/restaurant/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -33,13 +33,13 @@ public UserDto createUser(UserDto userDto) {
...
 
 			User user = new User(userDto);
 
 			return new UserDto(userRepo.save(user));
 		} else {
-			throw new BadRequestException("Email already exists");
+			throw new BadRequestException("Email already exists or invalid email pleas follow this structure user@gmail.com");
```
Comment (crack-byte) [inline src/main/java/com/restaurant/service/impl/UserServiceImpl.java]: check spelling

File: src/main/java/com/restaurant/service/impl/RatingServiceImpl.java [hunk 4]
```diff
@@ -32,12 +38,16 @@ public class RatingServiceImpl implements RatingService {
 	 * @see com.restaurant.dto.RatingDto
 	 */
 	@Override
-	public RatingDto createRating(RatingDto ratingDto, long id) {// rating dto
+	public RatingDto createRating(RatingDto ratingDto, long orderId,long userId) {// rating dto
```
Comment (crack-byte) [inline src/main/java/com/restaurant/service/impl/RatingServiceImpl.java]: update comment details for the method

**Review by crack-byte — CHANGES_REQUESTED** [review]

File: src/main/java/com/restaurant/controller/RatingController.java (hunk 1 above)
Comment (krishna-ku) [reply src/main/java/com/restaurant/controller/RatingController.java]: done

File: src/main/java/com/restaurant/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (krishna-ku) [reply src/main/java/com/restaurant/service/impl/UserServiceImpl.java]: Done

File: src/main/java/com/restaurant/service/impl/UserServiceImpl.java (hunk 3 above)
Comment (krishna-ku) [reply src/main/java/com/restaurant/service/impl/UserServiceImpl.java]: Done

File: src/main/java/com/restaurant/service/impl/RatingServiceImpl.java (hunk 4 above)
Comment (krishna-ku) [reply src/main/java/com/restaurant/service/impl/RatingServiceImpl.java]: Done

File: src/main/java/com/restaurant/entity/OrderItem.java [hunk 5]
```diff
@@ -13,23 +16,27 @@
 @Data
 @Entity
 @NoArgsConstructor
-public class OrderItem extends SuperClass {
+public class OrderItem {
```
Comment (crack-byte) [inline src/main/java/com/restaurant/entity/OrderItem.java]: why removed the superclass?

**Review by crack-byte — CHANGES_REQUESTED** [review]

File: src/main/java/com/restaurant/service/impl/RatingServiceImpl.java (hunk 4 above)
Comment (crack-byte) [reply src/main/java/com/restaurant/service/impl/RatingServiceImpl.java]: add orderId and userId param in method comment

**Review by crack-byte — CHANGES_REQUESTED** [review]

**Review by crack-byte — APPROVED** [review]

**Pull request merged**